astropy/units/core.py

```diff
@@ -95,8 +95,8 @@ def _normalize_equivalencies(equivalencies):
         else:
             raise ValueError(
                 "Invalid equivalence entry {0}: {1!r}".format(i, equiv))
-        if not (isinstance(funit, UnitBase) and
-                (isinstance(tunit, UnitBase) or tunit is None) and
+        if not (funit is Unit(funit) and
+                (tunit is None or tunit is Unit(tunit)) and
```

Comment:
See https://github.com/astropy/astropy/pull/1894/files#r8301896 -- the present PR is a follow-up on that, just adding the parsing. It also relates to item 3 in the list of outstanding questions for the function units (https://github.com/astropy/astropy/pull/1894/files#r8301896).